backend/src/services/health.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-console */
+import mongoose from 'mongoose';
+import { createTerminus } from '@godaddy/terminus';
+
+export const setUpHealthEndpoint = <T>(server: T) => {
+  const onSignal = () => {
+    console.log('Server is starting clean-up');
+    return Promise.all([
+      () => {
+        if (mongoose.connection && mongoose.connection.readyState == 1) {
+          mongoose.connection.close();
+          () => {
+            console.info('Database connection closed');
+          };
```

Comment:
First Promise.all has only 1 function and it's not `async () => {}`. It's not a problem but it will be better to use `new Promise((resolve,reject)=>{...})` if you need this function to be async.
Second thing is why not use `console.info(...)` directly rather than making it function, also is it even working? I'm doubting it will work it's just a function there is no call present in this function.

Here is what I mean:
![image](https://user-images.githubusercontent.com/8071263/208973758-0f10816f-a98f-4dd2-a91f-666eb243c8bd.png)

Reply:
Indeed, good catch as well! I will align. 🙇